dashboard/src/components/FunctionList/FunctionDeployButton.tsx

```diff
@@ -0,0 +1,241 @@
+  navigateToFunction: (n: string, ns: string) => Promise<any>;
+}
+
+interface IFunctionDeployButtonState {
+  functionSpec: IFunction["spec"];
+  modalIsOpen: boolean;
+  name: string;
+  namespace: string;
+  error?: string;
+}
+
+class FunctionDeployButton extends React.Component<
+  IFunctionDeployButtonProps,
+  IFunctionDeployButtonState
+> {
+  public state: IFunctionDeployButtonState = {
+    functionSpec: {
+      deps: "",
+      function: "",
+      handler: "hello.handler",
+      runtime: "nodejs6",
+      type: "HTTP",
+    },
+    modalIsOpen: false,
+    name: "",
+    namespace: "default",
+  };
+
+  public componentDidMount() {
+    const generatedName = Moniker.choose();
+    this.setState({
+      functionSpec: {
+        ...this.state.functionSpec,
+        function: this.generateFunction(
+          this.state.functionSpec.runtime,
+          this.state.functionSpec.handler,
+        ),
+        handler: `${generatedName}.handler`,
+      },
+      name: generatedName,
+    });
+  }
+
+  public render() {
+    const { functionSpec: f, name, namespace } = this.state;
+    return (
+      <div className="FunctionDeployButton">
+        <button className="button button-accent" onClick={this.openModal}>
+          Deploy New Function
+        </button>
+        <Modal
+          isOpen={this.state.modalIsOpen}
+          onRequestClose={this.closeModal}
+          contentLabel="Modal"
+        >
+          {this.state.error && (
+            <div className="container padding-v-big margin-v-big bg-action">{this.state.error}</div>
+          )}
+          <form onSubmit={this.handleDeploy}>
+            <div>
+              <label htmlFor="name">Name</label>
+              <input id="name" onChange={this.handleNameChange} value={name} required={true} />
+            </div>
+            <div>
+              <label htmlFor="namespace">Namespace</label>
+              <input
+                name="namespace"
+                onChange={this.handleNamespaceChange}
+                value={namespace}
+                required={true}
+              />
+            </div>
+            <div>
+              <label htmlFor="handler">Handler</label>
+              <input
+                name="handler"
+                onChange={this.handleHandlerChange}
+                value={f.handler}
+                required={true}
+              />
+            </div>
+            <div>
+              <label htmlFor="runtimes">Runtimes</label>
+              <select onChange={this.handleRuntimeChange} value={f.runtime}>
+                {Object.keys(Runtimes).map(r => (
+                  <option key={Runtimes[r]} value={Runtimes[r]}>
+                    {r}
+                  </option>
+                ))}
+              </select>
+            </div>
+            <div>
+              <label htmlFor="dependencies">Dependencies ({this.runtimeToDepsDescription()})</label>
```

Comment:
Just wondering if it should be possible to update the dependencies after the function is deployed.

Reply:
It should be, yes. It should also be possible to update the handler and other things on the spec. I'll create an issue.

Reply:
#196